discovery_engine_core/ai/kpe/src/lib.rs

```diff
@@ -15,11 +15,23 @@
 //! The KPE pipeline extracts key phrases from a sequence.
 //!
 //! See `examples/` for a usage example.
-#![cfg_attr(
-    doc,
-    forbid(rustdoc::broken_intra_doc_links, rustdoc::private_intra_doc_links)
-)]
+
 #![forbid(unsafe_op_in_unsafe_fn)]
+#![deny(
+    clippy::pedantic,
+    clippy::future_not_send,
+    noop_method_call,
+    rust_2018_idioms,
+    unused_qualifications,
+    unsafe_code
```

Comment:
we can only `deny(unsafe_code)` but not `forbid(unsafe_code)` as usual because the `ndarray::s!` macro expands an `allow(unsafe_code)` when we call it.